### ISSUE TITLE
UDP tests fail with Netty >= 4.1.37

### ISSUE THREAD

**Issue by violetagg** [reactor/reactor-netty#793]
https://travis-ci.org/reactor/reactor-netty/builds/551743903

```
reactor.netty.udp.UdpClientTest > smokeTest FAILED
    java.lang.AssertionError
        at org.junit.Assert.fail(Assert.java:86)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertTrue(Assert.java:52)
        at reactor.netty.udp.UdpClientTest.smokeTest(UdpClientTest.java:104)
reactor.netty.udp.UdpServerTests > portBindingException FAILED
    java.lang.AssertionError: illegal-success
        at org.junit.Assert.fail(Assert.java:88)
        at reactor.netty.udp.UdpServerTests.portBindingException(UdpServerTests.java:267)
```

**Comment by bsideup** [comment]
Maybe related:
https://github.com/netty/netty/issues/9212

**Comment by bsideup** [comment]
For `UdpClientTest.smokeTest`, here is the observation:
with .36, `server.address().getPort()` returns non-zero port (both Linux and Mac)
with .37 and higher, it always returns 0 on Linux, but non-zero on Mac

The test fails because it creates a client with `UdpClient.create().port(serverPort)`, and `serverPort=0`, means the client never connects to the started server.

/cc @normanmaurer

**Comment by normanmaurer** [comment]
Please open an issue in the netty issue tracker

**Comment by violetagg** [comment]
This issue is caused by https://github.com/netty/netty/commit/e17ce934da501788c3737631a02d02ee89ebf9df
PR is proposed https://github.com/netty/netty/pull/9436

**Comment by violetagg** [comment]
The fix will be in Netty 4.1.39.Final

**Comment by smaldini** [comment]
Fixed by https://github.com/reactor/reactor-netty/commit/db90c6d291b4f61a1d55a4ad5c14266a53149471
